File: ss20_sepm_qse_07-master/backend/src/main/java/at/ac/tuwien/sepm/groupphase/backend/entity/Task.java
package at.ac.tuwien.sepm.groupphase.backend.entity;

import at.ac.tuwien.sepm.groupphase.backend.types.TaskStatus;
import lombok.*;
import org.mapstruct.Mapper;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Getter
@Setter
@EqualsAndHashCode
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Task {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String description;

    @Column(nullable = false, columnDefinition = "TIMESTAMP")
    private LocalDateTime startTime;

    @Column(nullable = false, columnDefinition = "TIMESTAMP")
    private LocalDateTime endTime;

    @ManyToOne(fetch=FetchType.EAGER)
    private Employee assignedEmployee;

    @Column(nullable = false)
    private TaskStatus status;

    @Column(nullable = false)
    private boolean priority;

    @Column(nullable = false)
    private boolean event;

    @Lob
    @Column
    private String publicInfo;

    @Lob
    @Column
    private byte[] eventPicture;

    @OneToMany(mappedBy = "task", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<TaskComment> taskComments;
}

File: ss20_sepm_qse_07-master/frontend/src/app/dtos/task.ts
export class Task {
  constructor(public id: number,
              public title: string,
              public description: string,
              public startTime: string,
              public endTime: string,
              public assignedEmployeeUsername: string,
              public status: string,
              public subjectId: string,
              public subjectName: string,
              public animalTask: boolean,
              public priority: boolean = false,
              public event: boolean = false,
              public publicInfo: string = null,
              public eventPicture: string = null) {
  }
}
